Assets/Plugins/Source/Editor/PreProcessConfigConfirmation.cs

```diff
@@ -0,0 +1,155 @@
+﻿/*
+* Copyright (c) 2021 PlayEveryWare
+* 
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+* 
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+* 
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE.
+*/
+
+using UnityEditor.Build.Reporting;
+using UnityEditor.Build;
+using UnityEditor;
+using UnityEngine;
+using System.IO;
+
+class PreProcessConfigConfirmation : IPreprocessBuildWithReport
+{
+    public int callbackOrder { get { return 0; } }
+    public void OnPreprocessBuild(BuildReport report)
+    {
+        Debug.Log("PreProcessConfigConfirmation_Windows.OnPreprocessBuild for target " + report.summary.platform + " at path " + System.IO.Directory.GetCurrentDirectory());
+        BuildTarget target = report.summary.platform;
+
+        switch (target)
+        {
+            case BuildTarget.StandaloneWindows:
```

Comment:
You can simplify this copy/pasted switch to instead use a readonly dictionary (or you could create a local one) of the target and the config file, something like
```
Dictionary<BuildTarget, string> buildFileMap = new Dictionary<BuildTarget, string> {
  { BuildTarget.StandaloneWindows, "Assets/StreamingAssets/EOS/EpicOnlineServicesConfig.json" },
  { BuildTarget.StandaloneWindows64, "Assets/StreamingAssets/EOS/EpicOnlineServicesConfig.json" },
  ...
}
```
Then the whole function can be simplified to
```
if (System.IO.File.Exists(buildFileMap[target]))
{
    Debug.Log("PreProcessConfigConfirmation.OnPreprocessBuild for target " + target + " config file exists");
}
else
{
    Debug.LogError("PreProcessConfigConfirmation.OnPreprocessBuild for target " + target + " config file is missing");
    throw new BuildFailedException("Config file is missing, refer to above message");
}
```
and you can remove the extra functions at the end. In this way, when we need to add new platforms, we don't have to copy/paste a section, but add a new line to the dictionary (or uncomment a line).

Reply:
Also, you should also have a default check for the case where the target platform isn't covered with some error message for that too.

Reply:
A dictionary in C# is the same as a map in C++, just a different name.